### PR TITLE
use apache math libraries. delete custom gamma function computations

### PR DESCRIPTION
**Pull request by akiezun**
I deleted our custom gamma function implementation and replaced it with apache math. 

@yfarjoun please have a look. should be trivial

### CODE REVIEW

File: src/main/java/org/broadinstitute/hellbender/utils/MathUtils.java [hunk 1]
```diff
@@ -191,10 +198,6 @@ public static double log10sumLog10(final double[] log10p, final int start, final
         return maxValue + (sum != 1.0 ? Math.log10(sum) : 0.0);
     }
 
-    private static final double LOG1MEXP_THRESHOLD = Math.log(0.5);
```
Comment (yfarjoun) [inline src/main/java/org/broadinstitute/hellbender/utils/MathUtils.java]: What were these constants used for? could they be used in programs that still haven't been ported and you are now removing them?

File: src/main/java/org/broadinstitute/hellbender/utils/MathUtils.java [hunk 2]
```diff
@@ -320,16 +323,6 @@ public static int sum(final List<Integer> list ) {
         return sum;
     }
 
-    public static int countOccurrences(final boolean element, final boolean[] array) {
```
Comment (yfarjoun) [inline src/main/java/org/broadinstitute/hellbender/utils/MathUtils.java]: same as above

File: src/main/java/org/broadinstitute/hellbender/utils/MathUtils.java (hunk 1 above)
Comment (akiezun) [inline src/main/java/org/broadinstitute/hellbender/utils/MathUtils.java]: YAGNI
they were for the gamma

**Comment by yfarjoun** [comment]
:+1:

**Comment by akiezun** [comment]
thanks. will merge once test pass after rebasing

**Pull request merged**